pkg/codegen/python/gen.go

```diff
@@ -1335,6 +1339,23 @@ func (mod *modContext) genResource(res *schema.Resource) (string, error) {
 			fmt.Fprintf(w, "                pulumi.log.warn(\"\"\"%s is deprecated: %s\"\"\")\n", pname, escaped)
 		}
 
+		unwrapped := prop.Type
+		for {
+			cur := codegen.UnwrapType(unwrapped)
+			if cur == unwrapped {
+				break
+			}
+			unwrapped = cur
+		}
+
+		// Emit logic to configure schema.ObjectType args that have defaults and deprecation messages.
+		if objType, ok := unwrapped.(*schema.ObjectType); ok {
+			fmt.Fprintf(w, "            if isinstance(%s, dict):\n", InitParamName(prop.Name))
```

Comment:
Being as permissive as possible this condition should probably be flipped around. Check if the value is an instance of the Args type, and if not assume it's a dict like thing. That will allow users to pass in other values that behave like dicts life defaultdict or similar.

Reply:
Added commit addressing this. Will squash.